site/app/libraries/socket/Server.php

```diff
@@ -91,108 +84,42 @@ private function broadcast(ConnectionInterface $from, string $content, string $p
         }
     }
 
-    /**
-     * Connection object is the object stored against $this->clients for a given connection
-     * Connection object contains all details of the connection and can be used to push messages to client
-     *
-     * User_ID refers to the user_id of the user who is making a connection to the socket from a web page
-     * For example: if instructor has opened a socket connection from Forum threads page, user_id is instructor
-     */
-
-    /**
-     * Fetches Connection object/s of a given User_ID
-     * @param string $user_id
-     * @return bool|ConnectionInterface
-     */
-    private function getSocketClients(string $user_id) {
-        if (isset($this->users[$user_id])) {
-            return $this->users[$user_id];
-        }
-        else {
-            return false;
-        }
-    }
-
     /**
      * Fetches User_ID of a given socket Connection object
-     * @param ConnectionInterface $conn
-     * @return string|false
      */
-    private function getSocketUserID(ConnectionInterface $conn) {
-        if (isset($this->sessions[$conn->resourceId])) {
-            return $this->sessions[$conn->resourceId];
-        }
-        else {
-            return false;
-        }
+    private function getSocketUserID(ConnectionInterface $conn): ?string {
+        return $this->sessions[$conn->resourceId] ?? null;
     }
 
     /**
      * Sets Connection object associativity with User_ID
-     * @param string $user_id
-     * @param ConnectionInterface $conn
-     * @return void
      */
     private function setSocketClient(string $user_id, ConnectionInterface $conn): void {
         $this->sessions[$conn->resourceId] = $user_id;
-        if (array_key_exists($user_id, $this->users)) {
-            $this->users[$user_id][] = $conn;
-        }
-        else {
-            $this->users[$user_id] = [$conn];
-        }
-    }
-
-    /**
-     * Deletes Connection object associativity with User_ID
-     * @param ConnectionInterface $conn
-     * @return void
-     */
-    private function removeSocketClient(ConnectionInterface $conn): void {
```

Comment:
This function was only being called in one place (`onClose`), so just moved its code into that caller.